hoomd/hpmc/ExternalFieldHarmonic.h

```diff
@@ -0,0 +1,468 @@
+// Copyright (c) 2009-2021 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+#ifndef _EXTERNAL_FIELD_HARMONIC_H_
+#define _EXTERNAL_FIELD_HARMONIC_H_
+
+/*! \file ExternalFieldHarmonic.h
+    \brief Declaration and implementation of ExternalFieldHarmonic class
+*/
+
+#include "hoomd/Compute.h"
+#include "hoomd/HOOMDMPI.h"
+#include "hoomd/VectorMath.h"
+
+#include "ExternalField.h"
+
+#ifndef __HIPCC__
+#include <pybind11/pybind11.h>
+#endif
+
+namespace hoomd
+    {
+namespace hpmc
+    {
+
+template<class Shape> class ExternalFieldHarmonic : public ExternalFieldMono<Shape>
+    {
+    using ExternalFieldMono<Shape>::m_pdata;
+    using ExternalFieldMono<Shape>::m_exec_conf;
+    using ExternalFieldMono<Shape>::m_sysdef;
+
+    public:
+    //! Constructor
+    ExternalFieldHarmonic(std::shared_ptr<SystemDefinition> sysdef,
+                          pybind11::array_t<double> r0,
+                          Scalar k,
+                          pybind11::array_t<double> q0,
+                          Scalar q,
+                          pybind11::array_t<double> symRotations)
+        : ExternalFieldMono<Shape>(sysdef), m_k_translational(k), m_k_rotational(q)
+        {
+        setReferencePositions(r0);
+        setReferenceOrientations(q0);
+        setSymmetricallyEquivalentOrientations(symRotations); // TODO: check for identity?
```

Comment:
Reading the code, it appears that the user is required to include the identity in the list. This should be documented.

Reply:
Added.